### ISSUE TITLE
[BUG] GpuArrayExists  encounters a CudfException on an input partition consisting of just empty lists  

### ISSUE THREAD

**Issue by gerashegalov** [NVIDIA/spark-rapids#5108]
**Describe the bug**

When a partition consists of just rows of empty lists, the executor crashes with
```
22/03/31 17:58:14 WARN TaskSetManager: Lost task 3.0 in stage 0.0 (TID 3) (127.0.0.1 executor 0): ai.rapids.cudf.CudfException: cuDF failure at: ../src/copying/copy.cu:368: Boolean mask column must be the same size as rhs column
	at ai.rapids.cudf.ColumnView.ifElseSV(Native Method)
	at ai.rapids.cudf.ColumnView.ifElse(ColumnView.java:590)
	at com.nvidia.spark.rapids.GpuArrayExists.$anonfun$imputeFalseForEmptyArrays$4(higherOrderFunctions.scala:338)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.GpuArrayExists.withResource(higherOrderFunctions.scala:312)
	at com.nvidia.spark.rapids.GpuArrayExists.$anonfun$imputeFalseForEmptyArrays$3(higherOrderFunctions.scala:337)
```

This is related to https://github.com/rapidsai/cudf/issues/10556. 

**Steps/Code to reproduce bug**
To deterministically reproduce the bug use a single row with an empty array
```python
spark.createDataFrame(
    [
        [ [] ],
    ],
    'a array<int>'
).createOrReplaceTempView('df')

sql("SELECT a, exists(a, x -> x = 10) AS exits10 FROM df").show()
```

**Expected behavior**
Query should not crash and produce output with the number of output rows matching input rows with the value `false`.
```
+---+-------+
|  a|exits10|
+---+-------+
| []|  false|
+---+-------+
```

**Environment details (please complete the following information)**
 - Environment location: any
 - Spark configuration settings related to the issue: N/A

**Additional context**
https://github.com/rapidsai/cudf/issues/10556

**Comment by jlowe** [comment]
This should be fixed now that rapidsai/cudf#10876 has been merged.

**Comment by gerashegalov** [comment]
verified as fixed
